grouper/secret.py

```diff
@@ -0,0 +1,210 @@
+from enum import Enum
+from sqlalchemy.orm import Session  # noqa
+from typing import Any  # noqa
+
+from grouper.constants import SECRETS_ADMIN
+from grouper.fe.forms import SecretForm
+from grouper.group import get_all_groups
+from grouper.model_soup import Group
+from grouper.models.user import User  # noqa
+from grouper.user_group import get_groups_by_user
+from grouper.user_permissions import user_has_permission
+
+
+class SecretError(Exception):
+    """Base exception for all exceptions related to secrets. This should be used
+    by plugins when raising secret related exceptions as well.
+    """
+    pass
+
+
+class SecretRiskLevel(Enum):
+    """Risk levels that secrets can have"""
+    low = 1
+    medium = 2
+    high = 3
```

Comment:
The standard for python is to use lower case for enum members: https://docs.python.org/3/library/enum.html